### PR TITLE
5747 codecov

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

https://github.com/web3/web3.js/issues/5747

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -186,62 +192,3 @@ jobs:
                 path: /tmp
         -   run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
         -   run: yarn build:docs
-
-  coverage-tests-unit:
```
Comment (jdevcs) [inline .github/workflows/build.yml:190]: These are removed as we can now individually check packages coverage on codecov report page, and by removing these actions CI time will be saved.

**Comment by jdevcs** [comment]
Some coverage reports: 
per branch and packages: https://app.codecov.io/gh/web3/web3.js/tree/junaid%2F5747codecov/packages
per file at line level: https://app.codecov.io/gh/web3/web3.js/blob/junaid%2F5747codecov/packages/web3-utils/src/promise_helpers.ts

**Review by Muhammad-Altabba — APPROVED** [review]
Great suggestion!

**Review by luu-alex — APPROVED** [review]

**Comment by luu-alex** [comment]
https://github.com/web3/web3.js/blob/4.x/package.json#L70-L74 thanks @jdevcs , since we are removing the build script for coverage we can remove the coverage scripts i used in the packages

**Pull request merged**